contrib/win32/win32compat/w32log.c

```diff
@@ -114,17 +118,37 @@ openlog_file()
 		wcscat_s(ssh_cfg_path, _countof(ssh_cfg_path), __wprogdata); /* "%programData%" */
 		wcscat_s(ssh_cfg_path, _countof(ssh_cfg_path), L"\\ssh"); /* "%programData%\\ssh" */
 
+		wchar_t* tmp_identity = NULL;
+		if (strcmp(identity, "sftp-server") == 0) {
+			tmp_identity = utf8_to_utf16(identity);
+			if (!tmp_identity)
+				return;
+		}
+		else {
+			tmp_identity = malloc((wcslen(tail) - 4) * sizeof(wchar_t));
+			if (!tmp_identity)
+				return;
+			if (wcsncpy_s(tmp_identity, wcslen(tail) - 4, tail + 1, wcslen(tail) - 5) != 0)
```

Comment:
free(tmp_identity)

Reply:
The better way to handle memory is to add a cleaup goto block.
instead of return, have "goto cleaup".

In the cleanup block, check if the pointer is non-null then deallocate the memory.
This is more readable and maintainable in long run.